torch/csrc/distributed/c10d/init.cpp

```diff
@@ -142,6 +142,10 @@ PyObject* c10d_init(PyObject* _unused) {
       .def(
           "get_tensors",
           &::c10d::GradBucket::getTensors,
+          py::call_guard<py::gil_scoped_release>())
+      .def(
+          "get_tensor",
```

Comment:
I think we should have just one API that returns the entire list of tensors as I mentioned in a separate comment.